tradeexecutor/ethereum/uniswap_v2/uniswap_v2_routing.py

```diff
@@ -311,8 +321,92 @@ def make_multihop_trade(
+
+        swap_tx = get_swap_transactions(trade)
+        uniswap = self.mock_partial_deployment_for_analysis(web3, swap_tx.contract_address)
+
+        tx_dict = swap_tx.get_transaction()
+        receipt = receipts[HexBytes(swap_tx.tx_hash)]
+
+        input_args = swap_tx.get_actual_function_input_args()
+
+        result = analyse_trade_by_receipt(
+            web3,
+            uniswap=uniswap,
+            tx=tx_dict,
+            tx_hash=swap_tx.tx_hash,
+            tx_receipt=receipt,
+            pair_fee=trade.pair.fee,
+        )
+
+        ts = get_block_timestamp(web3, receipt["blockNumber"])
+
+        if isinstance(result, TradeSuccess):
+
+            # v3 path includes fee (int) as well
+            path = [a.lower() for a in result.path if type(a) == str]
+
+            if trade.is_buy():
+                assert path[0] == reserve.address, f"Was expecting the route path to start with reserve token {reserve}, got path {result.path}"
+
+                price = 1 / result.price
+
+                executed_reserve = result.amount_in / Decimal(10 ** reserve.decimals)
+                executed_amount = result.amount_out / Decimal(10 ** base_token_details.decimals)
+
+                # lp fee is already in terms of quote token
+                lp_fee_paid = result.lp_fee_paid
+            else:
+                # Ordered other way around
+                assert path[0] == base_token_details.address.lower(), f"Path is {path}, base token is {base_token_details}"
+                assert path[-1] == reserve.address
+
+                price = result.price
+
+                executed_amount = -result.amount_in / Decimal(10 ** base_token_details.decimals)
+                executed_reserve = result.amount_out / Decimal(10 ** reserve.decimals)
+
+                # convert lp fee to be in terms of quote token
+                lp_fee_paid = result.lp_fee_paid * float(price) if result.lp_fee_paid else None
+
+            assert (executed_reserve > 0) and (executed_amount != 0) and (price > 0), f"Executed amount {executed_amount}, executed_reserve: {executed_reserve}, price: {price}, tx info {trade.tx_info}"
+
+            # Mark as success
+            state.mark_trade_success(
+                ts,
+                trade,
+                executed_price=float(price),
+                executed_amount=executed_amount,
+                executed_reserve=executed_reserve,
+                lp_fees=lp_fee_paid,
+                native_token_price=0,  # won't fix
+                # TODO: Check unit conversion here
+                cost_of_gas=result.get_cost_of_gas(),
+            )
+        else:
+            # Trade failed
+            report_failure(ts, state, trade, stop_on_execution_failure)
+
+    def mock_partial_deployment_for_analysis(
+        self,
+        web3: Web3,
+        router_address: str
+    ) -> UniswapV2Deployment:
+        return mock_partial_deployment_for_analysis(web3, router_address)
```

Comment:
I think we can get rid of this method as well since it doesn't really provide any value, but can be done later